src/components/search/geoSearch/searchGeoForm.js

```diff
@@ -49,6 +50,30 @@ const searchGeoForm = (services) => {
         });
         mapBoxService.appendMapContent({selection: []});
 
+            $('.map-geo-btn').on('click', event => {
+                event.preventDefault();
+                if($('#map-zoom-to-setting').val()!= '') {
+                    savePreferences(
+                        {map_zoom : parseFloat($('#map-zoom-to-setting').val())}
+                    );
+                    $('#map-zoom-from-setting').val(parseFloat($('#map-zoom-to-setting').val()));
+                }
+                if($('#map-position-to-setting').val()!= '') {
+                    var centerRes = $('#map-position-to-setting').val();
+                    centerRes = centerRes.split('[');
+                    centerRes = centerRes[1].split(']');
+                    centerRes = centerRes[0].split(',');
+
+                    var lng = centerRes[0].split('"');
+                    lng= lng[1];
+                    var lat = centerRes[1].split('"');
+                    lat= lat[1];
+                    var res = [lng, lat];
```

Comment:
with simplified serialization :
```
latlng = $('#map-position-to-setting').val().split(';') ;
lat = latlng[0] ;
lng = latlng[1] ;
```
**Or better use json into unique "#map-to-settings"**